client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c

```diff
@@ -0,0 +1,821 @@
+    if (!buffer) {
+        return KAA_TCP_SOCK_IO_ERROR;
+    }
+    if (!access_point_test_info.auth_packet_written) {
+        if (buffer_size != (sizeof(CONNECT_HEAD)+sizeof(CONNECT_PACK))) {
+            return KAA_TCP_SOCK_IO_ERROR;
+        }
+        *bytes_written = 0;
+        if (!memcmp(buffer,CONNECT_HEAD,sizeof(CONNECT_HEAD))) {
+            if (!memcmp(buffer+sizeof(CONNECT_HEAD),CONNECT_PACK,sizeof(CONNECT_PACK))) {
+                access_point_test_info.auth_packet_written = true;
+                *bytes_written = sizeof(CONNECT_HEAD)+sizeof(CONNECT_PACK);
+                return KAA_TCP_SOCK_IO_OK;
+            }
+        }
+    } else if (!access_point_test_info.socket_disconnected_write) {
+        if (buffer_size != sizeof(DISCONNECT_MESSAGE)) {
+            return KAA_TCP_SOCK_IO_ERROR;
+        }
+        if (!memcmp(buffer,DISCONNECT_MESSAGE,sizeof(DISCONNECT_MESSAGE))) {
+            access_point_test_info.socket_disconnected_write = true;
+            *bytes_written = sizeof(DISCONNECT_MESSAGE);
+            return KAA_TCP_SOCK_IO_OK;
+        }
+    }
+
+    return KAA_TCP_SOCK_IO_ERROR;
+}
+
+kaatcp_error_t kaatcp_get_request_connect(const kaatcp_connect_t *message
+                                        , char *buf
+                                        , size_t *buf_size)
+{
+    if (message->protocol_name_length != KAA_TCP_NAME_LENGTH) {
+        return KAATCP_ERR_BAD_PARAM;
+    }
+    if (message->protocol_version != PROTOCOL_VERSION) {
+        return KAATCP_ERR_BAD_PARAM;
+    }
+    if (message->next_ptorocol_id != 0x3553c66f) {
+        return KAATCP_ERR_BAD_PARAM;
+    }
+    if (message->connect_flags != KAA_CONNECT_FLAGS) {
+        return KAATCP_ERR_BAD_PARAM;
+    }
+    if (message->keep_alive != (KEEPALIVE * 1.2)) {
+        return KAATCP_ERR_BAD_PARAM;
+    }
+    if (message->sync_request && message->sync_request_size == sizeof(CONNECT_PACK)) {
+        if (!memcmp(CONNECT_PACK,message->sync_request,message->sync_request_size)) {
+            *buf_size = 0;
+            memcpy(buf,CONNECT_HEAD,sizeof(CONNECT_HEAD));
+            *buf_size += sizeof(CONNECT_HEAD);
+            memcpy(buf+(*buf_size),message->sync_request,message->sync_request_size);
+            *buf_size += message->sync_request_size;
+            access_point_test_info.request_connect = true;
+            return KAATCP_ERR_NONE;
+        }
+    }
+    return KAATCP_ERR_BAD_PARAM;
+}
+
+kaatcp_error_t kaatcp_fill_connect_message(uint16_t keepalive, uint32_t next_protocol_id
+                                         , char *sync_request, size_t sync_request_size
+                                         , char *session_key, size_t session_key_size
+                                         , char *signature, size_t signature_size
+                                         , kaatcp_connect_t *message)
+{
+    if (keepalive != (KEEPALIVE * 1.2)) {
+        return KAATCP_ERR_BAD_PARAM;
+    }
+    if (next_protocol_id != 0x3553c66f) {
```

Comment:
The KAA_PLATFORM_PROTOCOL_ID constant from "kaa_platform_common.h" may be used instead.